loadtest/control/actions.go

```diff
@@ -745,6 +745,136 @@ func Reload(u user.User) UserActionResponse {
 	return UserActionResponse{Info: "page reloaded"}
 }
 
+func ReloadGQL(u user.User) UserActionResponse {
+	err := u.GetInitialDataGQL()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	prefs, err := u.Store().Preferences()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	var userIds []string
+	for _, p := range prefs {
+		switch {
+		case p.Category == model.PreferenceCategoryDirectChannelShow:
+			userIds = append(userIds, p.Name)
+		case p.Category == "group_channel_show":
+			if err := u.GetUsersInChannel(p.Name, 0, 8); err != nil {
+				return UserActionResponse{Err: NewUserError(err)}
+			}
+		}
+	}
+
+	var chanId string
+	if c, err := u.Store().CurrentChannel(); err == nil {
+		chanId = c.Id
+	} else if err != nil && err != memstore.ErrChannelNotFound {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	if chanId != "" {
+		// Marking the channel as viewed
+		_, err := u.ViewChannel(&model.ChannelView{
+			ChannelId:     chanId,
+			PrevChannelId: "",
+		})
+		if err != nil {
+			return UserActionResponse{Err: NewUserError(err)}
+		}
+	}
+
+	userId := u.Store().Id()
+
+	err = u.GetWebappPlugins()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	_, err = u.GetAllTeams(0, 50)
```

Comment:
I see that this is also hardcoded in the old Reload, but shouldn't we retrieve all teams if there are more than 50?

Reply:
Yes, but I think it's out of scope for this PR. Let's track it separately?

Reply:
That makes sense to me :)